app/assets/javascripts/components/buttons/generic-button-component.js

```diff
@@ -0,0 +1,41 @@
+ManageIQ.angular.app.component('genericButtonComponent', {
+  bindings: {
+    angularForm: '=',
+    modelCopy: '=',
+    model: '=',
+    redirectUrl: '=?',
+    newRecord: '<',
+    entity: '=?',
+    entityName: '=?',
+    onSaveClicked: '&?',
+    onAddClicked: '&?',
+  },
+  controllerAs: 'vm',
+  controller: genericButtonController,
+  templateUrl: '/static/buttons/generic_button_component.html.haml',
+});
+
+genericButtonController.$inject = ['miqService', '$window'];
+
+function genericButtonController(miqService) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    vm.saveable = miqService.saveable;
+
+    vm.resetClicked = function() {
```

Comment:
I'm sorry too, but I disagree with your above reasoning.

The idea behind a button component is two-fold -
- The main form components should not be dealing with buttons like Reset/Cancel that do not have any form logic per se. Not to mention the fact, that it's common logic across the board.
- This will cut down all the repetitiousness we have in the forms.
Practically every form has these methods -`resetClicked`...`cancelClicked`...`resetClicked`...`cancelClicked`.......
That is not a good design. Period.

The button component has a very high level knowledge that the model/modelCopy exist, and that's all.
It has no idea about the nitty-gritty details of the model.